labs-ios-starter/Surveys/View Controllers/TopicQuestionsViewController.swift

```diff
@@ -27,31 +115,41 @@ class TopicQuestionsViewController: UIViewController {
 
 extension TopicQuestionsViewController: UICollectionViewDataSource, UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return questions + 1
+        return questions.count
```

Comment:
Sorry, I should have documented these methods better. You did a pretty good job at figuring it out anyways. The + 1 logic, and switch statement below in cellForRowAt are used to dynamically increase the number of cells. So adding an extra cell here means that a cell will be added as the new last cell, you can press that to add a new questionCell to the collectionView.

Reply:
Just to confirm - Aren't we able to just add a question to the questions array which will reload the tableView and show the new question now that it's set up with an array?